### PR TITLE
Vendor `discord.ext.menus`

### PR DESCRIPTION
**Pull request by Drapersniper**
### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature

### Description of the changes
Add https://github.com/Rapptz/discord-ext-menus as a vendored module accessible to all Red users.
Currently on https://github.com/Rapptz/discord-ext-menus/commit/cc108bed812d0e481a628ca573c2eeeca9226b42

### CODE REVIEW

**Comment by Drapersniper** [comment]
Everything in #4021 should be addressed

File: README.md [hunk 1]
```diff
@@ -126,3 +126,6 @@ Red is named after the main character of "Transistor", a video game by
 
 Artwork created by [Sinlaire](https://sinlaire.deviantart.com/) on Deviant Art for the Red Discord
 Bot Project.
+
+This project vendors [discord.ext.menus](https://github.com/Rapptz/discord-ext-menus) package  made by Danny Y. (Rapptz) which is distributed under MIT License.
```
Comment (Jackenmen) [inline README.md]: Whitespace
```suggestion
This project vendors [discord.ext.menus](https://github.com/Rapptz/discord-ext-menus) package made by Danny Y. (Rapptz) which is distributed under MIT License.
```

File: README.md [hunk 2]
```diff
@@ -126,3 +126,6 @@ Red is named after the main character of "Transistor", a video game by
...
+Copy of this license can be found in [LICENSE](redbot/vendored/discord-ext-menus.LICENSE) file in [redbot/vendored/discord](redbot/vendored/discord) folder of this repository.
```
Comment (Jackenmen) [inline README.md]:
```suggestion
Copy of this license can be found in [discord-ext-menus.LICENSE](redbot/vendored/discord-ext-menus.LICENSE) file in [redbot/vendored](redbot/vendored) folder of this repository.
```

File: LICENSE [hunk 3]
```diff
@@ -700,3 +700,6 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
+
+This project vendors discord.ext.menus package (https://github.com/Rapptz/discord-ext-menus) made by Danny Y. (Rapptz) which is distributed under MIT License.
+Copy of this license can be found in LICENSE file in redbot/vendored folder of this repository.
```
Comment (Jackenmen) [inline LICENSE]:
```suggestion
Copy of this license can be found in discord-ext-menus.LICENSE file in redbot/vendored folder of this repository.
```

**Review by Jackenmen — DISMISSED** [review]
LGTM

**Review by Jackenmen — APPROVED** [review]
LGTM

**Pull request merged**